src/Constraint/BoundsProvidingInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver\Constraint;
+
+interface BoundsProvidingInterface
+{
+    /**
+     * Returns an array of bounds.
```

Comment:
Should return 2 integers, nothing more, a lower bound and an upper bound.

Reply:
No, that's not possible for MultiConstraint and != operators. Your goal is still perfectly achievable with this variant. I don't want to build a "good enough" solution here. We can still do that when writing the bounds into the autoload.php.

Reply:
Every constraint has exactly one upper and one lower boundary. That's what the meaning of the word boundary is. It's perfectly possible to do this, if you didn't try to calculate something other than boundaries in this PR.

If you don't want to build a tool for calculating upper/lower boundaries, I suggest you rename this and make a separate PR.